apycula/gowin_pack.py

```diff
@@ -392,11 +392,31 @@ def place(db, tilemap, bels, cst, args):
                         for brow, bcol in fuses:
                             tile[brow][bcol] = 1
 
-            if int(num) < 6:
+            if int(num) < 6 and int(parms['FF_USED'], 2):
                 mode = str(parms['FF_TYPE']).strip('E')
-                dffbits = tiledata.bels[f'DFF{num}'].modes[mode]
+                dff_attrs = set()
+                add_attr_val(db, 'SLICE', dff_attrs, cls_attrids['REGMODE'], cls_attrvals['FF'])
+                # REG0_REGSET and REG1_REGSET select set/reset or preset/clear options for each DFF individually
+                if mode in {'DFFR', 'DFFC', 'DFFNR', 'DFFNC'}:
+                    add_attr_val(db, 'SLICE', dff_attrs, cls_attrids[f'REG{int(num) % 2}_REGSET'], cls_attrvals['RESET'])
+                elif mode not in {'DFF', 'DFFN'}:
+                    add_attr_val(db, 'SLICE', dff_attrs, cls_attrids[f'REG{int(num) % 2}_REGSET'], cls_attrvals['SET'])
+                # are set/reset/clear/preset port needed?
+                if mode not in {'DFF', 'DFFN'}:
+                    add_attr_val(db, 'SLICE', dff_attrs, cls_attrids['LSRONMUX'], cls_attrvals['LSRMUX'])
+                # invert clock?
+                if mode in {'DFFN', 'DFFNR', 'DFFNC', 'DFFNP', 'DFFNS'}:
+                    add_attr_val(db, 'SLICE', dff_attrs, cls_attrids['CLKMUX_CLK'], cls_attrvals['INV'])
```

Comment:
Possibly you need here
```
else:
add_attr_val(db, 'SLICE', dff_attrs, cls_attrids['CLKMUX_CLK'], cls_attrvals['SIG'])
```

Reply:
Common sense dictates that the default values should not have fuses set, but in case we come across such a chip, so we add it.

Thanks;)